app/controllers/notes_controller.rb

```diff
@@ -45,6 +46,15 @@ def destroy
 
   private # Everything below this line is private
 
+  def authorize
+    authenticate_user!
+    if current_user.nil?
+      redirect_to root_path
+    else
+      @user = current_user
```

Comment:
setting the user here is not really necessary, isn't it?

Reply:
Very good catch. `@user` is never ever accessed from within in the notes controller or any of its views ...

Fixed in 10a30cce52bd6965150c4115f385d2aedd7c3c35.